frontend/src/components/common/Resource/Resource.tsx

```diff
@@ -894,3 +895,175 @@ export function ConditionsSection(props: { resource: KubeObjectInterface | null
     </SectionBox>
   );
 }
+
+export interface VolumeSectionProps {
+  volume?: KubeVolume;
+  resource: KubeObjectInterface | null;
+}
+
+export interface PrintVolumeLinkProps {
+  name?: string;
+  configMap?: {
+    name: string;
+    defaultMode: number;
+  };
+  persistentVolumeClaim?: {
+    claimName: string;
+    readOnly: boolean;
+  };
+}
+
+export function VolumeSection(props: VolumeSectionProps) {
+  const { resource } = props;
+
+  const volumes = resource?.spec?.volumes;
+
+  if (!volumes) {
+    return null;
+  }
+
+  const namespace = resource?.metadata?.namespace;
+
+  const { t } = useTranslation();
+
+  /*
+   * printVolumeLink will print a working link that is set within the router using fields from the resource as params
+   */
+  function printVolumeLink(volumeName: string, volume: PrintVolumeLinkProps) {
```

Comment:
Because these functions return components, they should be components (with capital letters, and taking a props as an argument).